django-stubs/contrib/admin/decorators.pyi

```diff
@@ -1,50 +1,53 @@
 from collections.abc import Callable, Sequence
-from typing import Any, TypeVar, overload
+from typing import Any, TypeVar, Union, overload
 
 from django.contrib.admin import ModelAdmin
 from django.contrib.admin.sites import AdminSite
 from django.db.models import Combinable, QuerySet
 from django.db.models.base import Model
 from django.db.models.expressions import BaseExpression
 from django.http import HttpRequest
-from django.utils.functional import _StrPromise
+from django.utils.functional import _StrOrPromise
+from typing_extensions import TypeAlias
 
-_T = TypeVar("_T")
 _Model = TypeVar("_Model", bound=Model)
 _ModelAdmin = TypeVar("_ModelAdmin", bound=ModelAdmin)
 _Request = TypeVar("_Request", bound=HttpRequest)
 _QuerySet = TypeVar("_QuerySet", bound=QuerySet)
+# This is deliberately different from _DisplayT defined in contrib.admin.options
+_DisplayCallable: TypeAlias = Union[Callable[[_ModelAdmin, _Model], Any], Callable[[_Model], Any]]  # noqa: Y037
```

Comment:
Need to suppress Y037. Can't use the pipe syntax here for some reason, otherwise [mypy-self-check (3.9)](https://github.com/typeddjango/django-stubs/actions/runs/3687912473/jobs/6242079013#logs) fails with:

> django-stubs/contrib/admin/decorators.pyi:18: error: Unsupported left operand type for | ("object")